egs/wsj/s5/steps/nnet3/components.py

```diff
@@ -213,6 +238,24 @@ def AddSigmoidLayer(config_lines, name, input, self_repair_scale = None):
     return {'descriptor':  '{0}_sigmoid'.format(name),
             'dimension': input['dimension']}
 
+def AddUnfoldedRnnLayer(config_lines, name, input, output_dim, ng_affine_options = "", self_repair_scale = None):
+    components = config_lines['components']
+    component_nodes = config_lines['component-nodes']
+
+    components.append("component name={0}_affine type=NaturalGradientAffineComponent input-dim={1} output-dim={2} {3}".format(name, input[0]['dimension'] + output_dim, output_dim, ng_affine_options))
+    self_repair_string = "self-repair-scale={0:.10f}".format(self_repair_scale) if self_repair_scale is not None else ''
+    components.append("component name={0}_tanh type=TanhComponent dim={1}".format(name, output_dim, self_repair_string))
+    # create a dummy node as a recurrent input to the first frame (it is expected to be 0 because it would not be computable in the computation graph) 
+    component_nodes.append("component-node name={0}_tanh_dummy component={0}_tanh input=IfDefined(ReplaceIndex({0}_tanh_t, x, -1))".format(name))
```

Comment:
It is to made the "dummy" node  NOT computable so that it would be a zero input to the first frame (just like how "IfDefined" descriptor is used in the RNN setting). Will add to the comments.